handlers.go

```diff
@@ -151,31 +151,33 @@ var CreateLogFilesHandler = Handler{
 			},
 		)
 
-		if m.Cfg.FifoLogWriter != nil {
-			if err := captureFifoToFile(m.logger, logFifoPath, m.Cfg.FifoLogWriter); err != nil {
-				m.logger.Warnf("captureFifoToFile() returned %s. Continuing anyway.", err)
-			}
-		}
-
 		m.logger.Debug("Created metrics and logging fifos.")
 
 		return nil
 	},
 }
 
-// BootstrapLoggingHandler is a named handler that will set up fifo logging of
+// createBootstrapLoggingHandler returns a named handler that will set up fifo logging of
 // firecracker process.
-var BootstrapLoggingHandler = Handler{
-	Name: BootstrapLoggingHandlerName,
-	Fn: func(ctx context.Context, m *Machine) error {
-		if err := m.setupLogging(ctx); err != nil {
-			m.logger.Warnf("setupLogging() returned %s. Continuing anyway.", err)
-		} else {
-			m.logger.Debugf("setup logging: success")
-		}
+func createBootstrapLoggingHandler(logFifo string) Handler {
+	return Handler{
+		Name: BootstrapLoggingHandlerName,
+		Fn: func(ctx context.Context, m *Machine) error {
+			if err := m.setupLogging(ctx); err != nil {
+				m.logger.Warnf("setupLogging() returned %s. Continuing anyway.", err)
```

Comment:
If `setupLogging` returns an error, we should probably return from the function (even if we only log the error and return nil). Otherwise, we can continue on from here to the `captureFifoToFile` call below and get blocked again due to `setupLogging` failing.

Is there any expected reason for `setupLogging` to fail? Would it just be simpler to return any error from it?

Reply:
Fixed! Discussed offline but what was decided was to return nil if an error occurs, that we avoid the file potentially being blocked by open.